tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -421,7 +431,9 @@ SpanBuffer SwapBuffers()
                         // this will override the TraceContext sampling priority when we do a SpanBuffer.TryWrite
                         chunkSamplingPriority = SamplingPriorityValues.UserKeep;
                         Interlocked.Increment(ref _droppedP0Traces); // increment since we are sampling out the entire trace
-                        Interlocked.Add(ref _droppedP0Spans, spans.Count - singleSpanSamplingSpans.Count);
+                        var spansDropped = spans.Count - singleSpanSamplingSpans.Count;
+                        Interlocked.Add(ref _droppedP0Spans, spansDropped);
+                        TelemetryMetrics.Instance.Record(Count.SpanDropped, MetricTags.DropReason_SingleSpanSampling, spansDropped);
```

Comment:
Actually SingleSpanSampling is the reason why we keep some spans not why we drop them. 
Sampling is always the reason we drop the spans, we just keep them sometimes because of erros, Single Span Sampling... 
The only solution I see to report this well would be to have a `spans.kept` metric. But maybe that's overkill?

Reply:
I agree with your concerns. Additionally, no one in .NET as stats computation enabled for now, so clearly not a big concern for us now. 
I think the only thing we should change is the tag (and use the sampling one). Cause saying the reason is singlespansampling is weird.

Reply:
Except this _is_ to do with singlespansampling right, i.e. the spans we dropped were dropped because they _weren't_ sampled, and without this we have _no_ metrics about singlespansampling. 

I guess I'm trying to wedge a square peg into a round hole here 😄

Reply:
> I guess I'm trying to wedge a square peg into a round hole here 😄

It's basically what I thought when I read this :) Maybe you can change the tag name to something like `PartiallySampledAndKeptWithSingleSpanSampling`.  But `DropReason_SingleSpanSampling` is wrong.